@here/harp-mapview/lib/MapView.ts

```diff
@@ -2574,18 +2571,63 @@ export class MapView extends THREE.EventDispatcher {
         this.m_screenCollisions.update(width, height);
 
         this.m_pixelToWorld = undefined;
+        this.m_fog.update(this, this.m_viewRanges.maximum);
 
+        this.computeTargetAndDistance();
+        this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(this, this.m_targetDistance);
```

Comment:
Here I would strongly suggest to provide interface for zoom level calculation i.e.:
ZoomLevelEvaluator.
By default it would be the one based on target distance, but we could of course provide our own and customers will have free hand to define own implementation - for example ground distance based - as in other 3D engines.

Reply:
Actually I think we need an interface for computing the distance b/c converting distance to zoomlevel will never change.
As I mentioned in another comment my goal is to create a proper `MapViewState` object and maybe we could have an interface that computes the whole `MapViewState`

Reply:
Or giving it a second thought that might be a bad idea b/c the whole `MapViewState` exercise is to reduce our public API. If we have that interface `MapViewState` would be public again.
Anyways: I think this is out of scope for this task but we should keep it in mind.